java/code/src/com/suse/manager/webui/controllers/ProductsController.java

```diff
@@ -263,6 +271,54 @@ public static String addProduct(Request request, Response response, User user) {
         return json(response, resultMap);
     }
 
+    /**
+     * Add optional channels to be synced in the SUSE Manager Server
+     *
+     * @param request the request
+     * @param response the response
+     * @param user the user
+     * @return the map of requested channels with a success/failed flag in a JSON format
+     */
+    public static String addOptionalChannels(Request request, Response response, User user) {
+        List<String> channels = Json.GSON.fromJson(request.body(), new TypeToken<List<String>>() { }.getType());
+        if (!user.hasRole(RoleFactory.SAT_ADMIN)) {
+            throw new IllegalArgumentException("Must be SAT_ADMIN to synchronize products");
+        }
```

Comment:
As I understand it, `withProductAdmin()` wrapper already checks for `SAT_ADMIN` role before calling any of these route methods. So in effect, this piece is unreachable.

I think we can safely remove this if clause from here and the other methods in this controller.